typescript/packages/jumble/src/components/CommandCenter.tsx

```diff
@@ -0,0 +1,404 @@
+import { Command } from "cmdk";
+import { useState, useEffect, useRef } from "react";
+import "./commands.css";
+import { castNewRecipe, Charm, CharmManager, iterate } from "@commontools/charm";
+import { useCharmManager } from "@/contexts/CharmManagerContext";
+import { useMatch, useNavigate } from "react-router-dom";
+import { castSpell } from "@/search";
+import { charmId } from "@/utils/charms";
+import { Dialog, VisuallyHidden } from "radix-ui";
+import { NAME } from "@commontools/builder";
+import { DocImpl, getRecipe } from "@commontools/runner";
+
+export function CommandCenter() {
```

Comment:
This will need to be broken into subcomponents in short order and have a state machine to handle more advanced flows. I have a pretty clear idea of what that looks like, just planting a flag for now.